docs/docs/unit-testing.md

```diff
@@ -0,0 +1,257 @@
+---
+title: Unit testing
+---
+
+Unit testing is a great way to protect against errors in your code before you
+deploy it. While Gatsby does not include support for unit testing out of the
+box, it only takes a few steps to get up and running. However there are a few
+features of the Gatsby build process that mean the standard Jest setup doesn't
+quite work. This guide shows you how to set it up.
+
+## Setting up your environment
+
+The most popular testing framework for React is [Jest](https://jestjs.io/),
+which was created by Facebook. While Jest is a general purpose JavaScript unit
+testing framework, it has lots of features that make it work particularly well
+with React.
+
+For this guide, you will be starting with `gatsby-starter-blog`, but the
+concepts should be the same or very similar for your site.
+
+First you need to install Jest and some more required packages. You need to
```

Comment:
Is this actually necessary with v2 since Gatsby is using Babel 7 already? Seems you would only need to install the presets and plugins? Also, looks like presets are missing from install step unless they are already present for Gatsby itself.

Reply:
Good point: the presets are missing, and do need installing. I'll add them. Even though, as you say, Babel 7 is installed by Gatsby, it seems that unless it is added as a dependency of the project, the wrong version is picked-up. Likewise plugins are not found unless specifically included. All my testing was done on the latest beta.

Reply:
As for your point about whether this is required: yes, it is, as jest doesn't know that it needs to use babel, as there's no .babelrc. You could add one, but of course that is then also used in the Gatsby build. Using a Jest preprocessor seems to be the best way (and is the way used in the Gatsby codebase). I tried to stick as close as possible to the way it's done in the Gatsby site itself.

Reply:
Oh yeah, using preprocessor for sure! I meant having to install babel 7 since it's already used by gatsby, but seems that it is in fact needed!